sources/bloodhound/src/bin/bottlerocket-checks/checks.rs

```diff
@@ -0,0 +1,1280 @@
+/// `modprobe -n -v udf` will return a line matching 'install /bin/true` in its
+/// output.
+fn check_modprobe(result: &mut results::CheckerResult) {
+    if result.status == results::CheckStatus::FAIL {
+        return;
+    }
+
+    if let Some(found) =
+        look_for_string_in_output(MODPROBE_CMD, ["-n", "-v", "udf"], "install /bin/true")
+    {
+        if !found {
+            result.error = "modprobe for udf is not disabled".to_string();
+            result.status = results::CheckStatus::FAIL;
+        } else {
+            result.status = results::CheckStatus::PASS;
+        }
+    } else {
+        result.error = "unable to parse modprobe output to check if udf is enabled".to_string();
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct BR01030100Checker {}
+
+impl results::Checker for BR01030100Checker {
+    fn execute(&self) -> results::CheckerResult {
+        let mut result = results::CheckerResult {
+            error: String::new(),
+            status: results::CheckStatus::SKIP,
+        };
+
+        let mut enabled = true;
+        let mut skipped = false;
+        // This isn't the most efficient because it results in the proc file being opened, read, and closed three times.
+        // It is a single line, so there shouldn't be too much extra overhead compared to the simplicity of doing it
+        // this way, but if needed this should be changed to read the contents once, then check each expected value in
+        // the same content string.
+        if let Some(found) = look_for_string_in_file(PROC_CMDLINE_FILE, "dm-mod.create=root,,,ro,0")
+        {
+            if !found {
+                enabled = false;
+            }
+        } else {
+            skipped = true;
+        }
+
+        if let Some(found) = look_for_string_in_file(PROC_CMDLINE_FILE, "root=/dev/dm-0") {
+            if !found {
+                enabled = false;
+            }
+        } else {
+            skipped = true;
+        }
+
+        if let Some(found) = look_for_string_in_file(PROC_CMDLINE_FILE, "restart_on_corruption") {
+            if !found {
+                enabled = false;
+            }
+        } else {
+            skipped = true;
+        }
+
+        if skipped {
+            result.error = "unable to verify cmdline includes dm-verity settings".to_string();
+        } else if enabled {
+            result.status = results::CheckStatus::PASS;
+        } else {
+            result.error = "unable to verify dm-verity enforcement, settings not found".to_string();
+            result.status = results::CheckStatus::FAIL;
+        }
+
+        result
+    }
+
+    fn metadata(&self) -> results::CheckerMetadata {
+        results::CheckerMetadata {
+            title: "Ensure dm-verity is configured".to_string(),
+            id: "1.3.1".to_string(),
+            level: 1,
+            name: "br01030100".to_string(),
```

Comment:
Would it be worth naming these checks something like `br_1_3_1` rather than using zeroes as placeholders?

You could also use a proc macro to derive a method that would turn the struct name into the ID and the name.

Reply:
The name needs the zero padding to allow for easy human expected sorting of the results. But a macro may help with some of this boilerplate. I'll look into that.

Reply:
I couldn't think of a good way to improve on this with a macro, but happy to be told otherwise.

The problem I see is with needing the title, ID #, level, and prefix (e.g. `br`), having a macro that would take the necessary input wouldn't be any more concise than just having it stated here, and there would be the overhead of the macro expansion and any manipulation of the ID and prefix to emit the desired field values.